test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +90,9 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(inputCommand.equals("add") || inputCommand.equals("clear") || inputCommand.equals("delete")){
```

Comment:
Can you reduce the amount of code and improve readability by using the `isMutating()` method?